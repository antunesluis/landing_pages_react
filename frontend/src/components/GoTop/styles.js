import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    position: fixed;
    display: flex;
    align-items: center;
    justify-content: center;
    width: 4rem;
    height: 4rem;
    bottom: 2rem;
    right: 2rem;
    z-index: 6;

    background: ${theme.colors.primaryColor};
    color: ${theme.colors.white};
    border-radius: 0.4rem;
    transition: all 0.3s ease-in-out;

    &:hover,
    &:focus {
      background: ${theme.colors.secondaryColor};
      color: ${theme.colors.white};
      transform: scale(1.05);
    }
  `}
`;
