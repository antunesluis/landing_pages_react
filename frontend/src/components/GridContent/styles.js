import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${() => css`
    text-align: center;
    max-width: 58rem;
    margin: 0 auto;
  `}
`;

export const HtmlContent = styled.div`
  ${({ theme }) => css`
    margin: ${theme.spacings.xhuge} 0;

    // Estilização para textos e parágrafos
    p {
      font-size: ${theme.fonts.sizes.medium};
      line-height: 1.5;
      margin: ${theme.spacings.medium} 0;
    }

    // Estilização para cabeçalhos dentro do conteúdo
    h1,
    h2,
    h3,
    h4,
    h5,
    h6 {
      font-weight: 700;
      font-family: ${theme.fonts.family.secondary};
    }

    h1 {
      font-size: ${theme.fonts.sizes.xlarge};
      margin: ${theme.spacings.xlarge} 0;
    }

    h2 {
      font-size: ${theme.fonts.sizes.large};
      margin: ${theme.spacings.large} 0 ${theme.spacings.medium};
    }

    h3 {
      font-size: ${theme.fonts.sizes.medium};
      margin: ${theme.spacings.xlarge} 0 ${theme.spacings.small};
    }

    // Estilização para links
    a {
      color: ${theme.colors.secondaryColor};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      &:hover {
        filter: brightness(50%);
      }
    }

    // Estilização para listas
    ul,
    ol {
      margin: ${theme.spacings.small} 0;
      padding: ${theme.spacings.medium} ${theme.spacings.large};
    }

    ul {
      list-style: disc;
    }

    ol {
      list-style: decimal;
    }

    li {
      margin: ${theme.spacings.small} 0;
      font-size: ${theme.fonts.sizes.medium};
      text-align: left;
    }

    // Estilização para imagens
    img {
      max-width: 100%;
      height: auto;
    }

    // Estilização para citações
    blockquote {
      border-left: 0.5rem solid ${theme.colors.secondaryColor};
      color: ${theme.colors.primaryColor};
      margin: ${theme.spacings.medium} 0;
      padding: ${theme.spacings.medium} ${theme.spacings.large};
      font-style: italic;
    }

    // Estilização para tabelas
    table {
      width: 100%;
      border-collapse: collapse;
      margin: ${theme.spacings.medium} 0;
    }

    table td,
    table th {
      padding: ${theme.spacings.small};
      border: 1px solid ${theme.colors.mediumGray};
      text-align: left;
    }

    table th {
      background: ${theme.colors.primaryColor};
      color: ${theme.colors.white};
    }

    // Estilização para elementos de ênfase
    strong,
    b {
      font-weight: bold;
    }

    em,
    i {
      font-style: italic;
    }

    /*
    @media ${theme.media.lteMedium} {
      h1 {
        font-size: ${theme.fonts.sizes.large};
      }
      h2 {
        font-size: ${theme.fonts.sizes.medium};
      }
      h3 {
        font-size: ${theme.fonts.sizes.small};
      }
      padding: ${theme.spacings.medium};
    } */
  `}
`;
