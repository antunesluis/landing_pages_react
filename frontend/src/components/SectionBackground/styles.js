import styled, { css } from 'styled-components';

const containerBackgroundActivate = (theme) => css`
  background-image: url(${theme.backgroundImage});
  background-size: cover;
  background-position: center center;
  background-repeat: no-repeat;
  color: ${theme.colors.white};

  &::before {
    content: '';
    position: absolute;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    background: rgba(0, 0, 0, 0.6);
    z-index: 1;
  }

  & > * {
    position: relative;
    z-index: 2;
  }
`;

export const Container = styled.div`
  ${({ theme, background }) => css`
    background: ${theme.colors.white};
    color: ${theme.colors.primaryColor};

    ${background && containerBackgroundActivate(theme)};

    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
