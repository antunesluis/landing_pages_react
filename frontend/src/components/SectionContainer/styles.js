import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    max-width: ${theme.spacings.maxWidth};
    margin: 0 auto;
    padding: ${theme.spacings.large};
  `}
`;
