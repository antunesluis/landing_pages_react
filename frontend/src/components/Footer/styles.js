import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;

    a {
      color: inherit;
      text-decoration: none;
      font-size: ${theme.fonts.sizes.small};
    }

    > ${SectionContainer} {
      padding: 0;
    }
  `}
`;
