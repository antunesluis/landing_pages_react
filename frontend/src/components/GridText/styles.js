import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextComponent } from '../TextComponent/styles';

export const Container = styled.div`
  ${({ theme }) => css`
    width: 100%;
    max-width: ${theme.spacings.maxWidth};
    margin: 0 auto;

    ${TextComponent} {
      margin-bottom: ${theme.spacings.xhuge};
      max-width: 70ch;
      opacity: 0.9;
    }

    > ${HeadingContainer} {
      text-shadow: 0.1rem 0.1rem 0.2rem rgba(0, 0, 0, 0.2);
      margin-bottom: ${theme.spacings.medium};
    }

    @media ${theme.media.lteMedium} {
      ${TextComponent} {
        margin-bottom: ${theme.spacings.large};
      }
    }
  `}
`;

export const Grid = styled.div`
  ${({ theme }) => css`
    counter-reset: grid-counter;
    display: grid;
    max-width: 100%;
    grid-template-columns: repeat(auto-fit, minmax(min(100%, 32rem), 1fr));
    gap: ${theme.spacings.xlarge};
  `}
`;

export const GridElement = styled.div`
  ${({ theme }) => css`
    position: relative;
    border-radius: 8px;
    background: rgba(255, 255, 255, 0.03);
    transition: all 0.3s ease-in-out;
    padding: ${theme.spacings.medium};

    &:hover {
      transform: translateY(-6px);
      background: rgba(255, 255, 255, 0.05);
    }

    ${HeadingContainer} {
      position: relative;
      left: 5rem;
      margin-bottom: ${theme.spacings.small};

      // Evita que o texto encoste na lateral direita
      padding-right: ${theme.spacings.medium};
      // Garante que o heading n√£o ultrapasse o container
      max-width: calc(100% - 5rem);

      @media ${theme.media.lteMedium} {
        left: 4rem;
        max-width: calc(100% - 4rem);
      }
    }

    ${HeadingContainer}::before {
      counter-increment: grid-counter;
      content: counter(grid-counter);
      position: absolute;
      font-size: clamp(5rem, 8vw, 7.5rem);
      font-weight: 700;
      opacity: 0.15;
      top: -3rem;
      left: -6rem;
      transform: rotate(8deg);
      transition: all 0.2s ease-in-out;
    }

    &:hover ${HeadingContainer}::before {
      opacity: 0.2;
      transform: rotate(5deg) scale(1.1);
    }

    ${TextComponent} {
      line-height: 1.6;
      opacity: 0.85;
      padding-right: ${theme.spacings.small};
      max-width: 100%;
    }
  `}
`;
