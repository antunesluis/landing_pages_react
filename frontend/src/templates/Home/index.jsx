import { useEffect, useRef, useState } from 'react';
import { mapData } from '../../api/map-data';

import { Base } from '../Base';
import { PageNotFound } from '../PageNotFoud';
import { Loading } from '../Loading';

import { GridTwoColumn } from '../../components/GridTwoColumn';
import { GridContent } from '../../components/GridContent';
import { GridText } from '../../components/GridText';
import { GridImage } from '../../components/GridImage';
import { pageService } from '../../api/services';
import { useLocation } from 'react-router-dom';

function Home() {
  const [data, setData] = useState([]);
  const location = useLocation();
  const isMounted = useRef(true);

  useEffect(() => {
    const load = async () => {
      const pathName = location.pathname.replace(/[^a-z0-9-_]/gi, '');
      const slug = pathName ? pathName : 'landing-page';

      try {
        const json = await pageService.getPageBySlug(slug);
        const pageData = mapData(json.data);

        setData(() => pageData[0]);
      } catch {
        setData(undefined);
      }
    };

    if (isMounted.current === true) {
      load();
    }

    return () => {
      isMounted.current = false;
    };
  }, [location]);

  if (data === undefined) {
    return <PageNotFound />;
  }

  if (data && !data.slug) {
    return <Loading />;
  }

  const { menu, sections, footerHtml, slug } = data;
  const { links, text, link, srcImg } = menu;

  return (
    <Base
      links={links}
      footerHtml={footerHtml}
      logoData={{ text, link, srcImg }}
    >
      {sections.map((section, index) => {
        const { component } = section;
        const key = `${slug}-${index}`;

        if (component === 'section.section-two-columns') {
          return <GridTwoColumn key={key} {...section} />;
        }

        if (component === 'section.section-content') {
          return <GridContent key={key} {...section} />;
        }

        if (component === 'section.section-grid-text') {
          return <GridText key={key} {...section} />;
        }

        if (component === 'section.section-grid-image') {
          return <GridImage key={key} {...section} />;
        }
      })}
    </Base>
  );
}

export default Home;
