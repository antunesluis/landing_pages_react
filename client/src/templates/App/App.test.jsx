import { renderTheme } from '../../styles/render-theme';
import { screen } from '@testing-library/react';
import Home from '.';
import { theme } from '../../styles/theme';

test('renders heading with text Hello', () => {
  const { debug } = renderTheme(<Home />);
  const heading = screen.getByRole('heading', { name: 'Hello' });

  expect(heading).toBeInTheDocument(); // Valida que o heading est√° no DOM
  expect(heading).toHaveStyle({ background: theme.colors.primary });
  expect(heading).toMatchSnapshot(); // Cria um snapshot do heading

  debug(); // Inspeciona o DOM renderizado
});
