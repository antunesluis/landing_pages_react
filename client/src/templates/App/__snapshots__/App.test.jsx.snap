// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders heading with text Hello 1`] = `
<h1>
  Hello
</h1>
`;
